Geometry/CommonTopologies/interface/SimplePixelTopology.h

```diff
@@ -380,8 +380,10 @@ namespace pixelTopology {
 
     static constexpr uint16_t numberOfModules = 3892;
 
-    // 1024 bins, 10 bits
-    static constexpr uint16_t clusterBinning = 1024;
+    // 1000 bins (<10 bits) must be:
+    // - < 32*32 (warpSize*warpSize for block prefix scan for CUDA)
+    // - > in y of any module (in each y bin only pixels with same integer y)
```

Comment:
... but I don't understand the comment about `y` here :-/
What does
```c++
    // - > in y of any module
```
mean ?

Reply:
Right, let me explain better. I was referring to the local coordinate in the module so with `y` I mean the columns of the pixel grid of the module (so the local `y` of any pixel, an integer). So the binning should be  greater the number of columns in any module (for Phase1 the number of columns is constant, for Phase2 no). This is due to the fact that in clustering we give for granted that in each bin we only have the pixel belonging to the same column. See https://github.com/cms-sw/cmssw/blob/b3830cced4787ed0573f42a7fa42644607013803/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h#L325-L347

Therefore we need at least as many bins as number of columns (y). I can rephrase it in the comments as I did here.

Reply:
Also, I could tailor it to be nearer to the maximum of columns (I don't remember it by hearth at the moment).

Reply:
Ah, thanks !
Yes, I think if you just write "number of columns (y) in any module" it should be enough to understand.
Of course, feel free to write a longer explanation, as you did here, which is also good :-)

Reply:
> Also, I could tailor it to be nearer to the maximum of columns (I don't remember it by hearth at the moment).

If it has a measurable impact on the memory used, sure.